lib/rescue_from_duplicate/active_record/extension.rb

```diff
@@ -24,22 +24,28 @@ def _rescue_from_duplicate_handlers
 
     def create_or_update(*params, &block)
       super
-    rescue ActiveRecord::RecordNotUnique => exception
-      handler = exception_handler(exception)
+    rescue ActiveRecord::RecordNotUnique, ActiveRecord::StatementInvalid => exception
```

Comment:
We can also just bump rescue_form_duplicate and change the dependency constraints.